opm/models/blackoil/blackoilnewtonmethodparams.cpp

```diff
@@ -0,0 +1,81 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+
+#include <config.h>
+#include <opm/models/blackoil/blackoilnewtonmethodparams.hpp>
+
+#include <opm/models/utils/parametersystem.hpp>
+
+namespace Opm {
+
+template<class Scalar>
+void BlackoilNewtonParams<Scalar>::registerParameters()
+{
+    Parameters::Register<Parameters::DpMaxRel<Scalar>>
+        ("Maximum relative change of pressure in a single iteration");
+    Parameters::Register<Parameters::DsMax<Scalar>>
+        ("Maximum absolute change of any saturation in a single iteration");
+    Parameters::Register<Parameters::PriVarOscilationThreshold<Scalar>>
+        ("The threshold value for the primary variable switching conditions "
+                                                                        "after its meaning has switched to hinder oscilations");
```

Comment:
Unexpected indentation here.  Is that an accident?  On a tangential note, maybe we could fix the misprint ("oscilation" ➡️ "oscillation") as well, possibly in follow-up work?

Reply:
not on purpose, just me not paying attention to IDE autoindent. fixed this and the typo.